File: Graph/Graph.py
class Graph:
    def __init__(self, vertices):
        self.vertices = vertices
        self.adjacency_list = [[] for _ in range(vertices)]

    def add_edge(self, source, destination):
        self.adjacency_list[source].append(destination)
        self.adjacency_list[destination].append(source)

    def bfs(self, start_vertex):
        visited = [False] * self.vertices
        queue = Queue()

        visited[start_vertex] = True
        queue.enqueue(start_vertex)

        while not queue.is_empty():
            current_vertex = queue.dequeue()
            print(current_vertex, end=" ")

            for neighbor in self.adjacency_list[current_vertex]:
                if not visited[neighbor]:
                    visited[neighbor] = True
                    queue.enqueue(neighbor)

    def dfs(self, start_vertex):
        visited = [False] * self.vertices
        stack = Stack()

        stack.push(start_vertex)

        while not stack.is_empty():
            current_vertex = stack.pop()

            if not visited[current_vertex]:
                visited[current_vertex] = True
                print(current_vertex, end=" ")

                for neighbor in self.adjacency_list[current_vertex]:
                    stack.push(neighbor)

File: Graph/Graph.cs
using System.Collections.Generic;

namespace Struct.BFS
{
    class Graph
    {
        private List<int>[] _adjacencyList;
        private readonly int _v;

        public Graph(int v)
        {
            _v = v;
            _adjacencyList = new List<int>[_v];
            for (int i = 0; i < _v; i++)
            {
                _adjacencyList[i] = new List<int>();
            }
        }

        public void AddEdge(int v, int w)
        {
            _adjacencyList[v].Add(w);
        }

        public List<int> GetAdjacencyList(int v)
        {
            return _adjacencyList[v];
        }

        public int GetVerticesCount()
        {
            return _v;
        }
    }
}